File: 2766-find-the-prefix-common-array-of-two-arrays/2766-find-the-prefix-common-array-of-two-arrays.java
class Solution {
    public int[] findThePrefixCommonArray(int[] A, int[] B) {
        int n1 = A.length;
        int n2 = B.length;
        int[] C = new int[n1];
        
        for(int i=0; i<n1; i++){
            int count = 0;
            for(int j=0; j<=i; j++){
                for(int k=0; k<=i; k++){
                    if(A[j] == B[k]){
                        count++;
                        break;
                    }
                }
            }
            C[i] = count;
        }
        return C;
    }
}

File: 2766-find-the-prefix-common-array-of-two-arrays/2766-find-the-prefix-common-array-of-two-arrays.cpp
class Solution {
public:
    vector<int> findThePrefixCommonArray(vector<int>& nums1, vector<int>& nums2) {
        unordered_map<int,int>mpp;
        int c=0;
        vector<int> ans;
        for(int i=0;i<nums2.size();i++){
            mpp[nums1[i]]++;
            if(mpp[nums1[i]]==2) c++;
            mpp[nums2[i]]++;
            if(mpp[nums2[i]]==2) c++;
            ans.push_back(c);
        }
        return ans;
    }
};